project/Build.scala

```diff
@@ -1303,7 +1303,7 @@ object Build {
           val option = mapSourceURISetting(
               (artifactPath in fetchScalaSource).value,
               s"https://raw.githubusercontent.com/scala/scala/v${scalaVersion.value}/src/library/")
-          option +: prev
+          s"-P:scalajs:$option" +: prev
```

Comment:
Instead of just fixing this here, I suggest the following:

Add:

```scala
def scalaJSCompilerOption(o: String): Seq[String] =
  if (isGeneratingForIDE) Nil
  else Seq(s"-P:scalajs:$o")
```

And replace all usages of `addScalaJSCompilerOption` with that function and the relevant sbt incantation.

(e.g. `scalacOptions ++= scalaJSCompilerOption("fixClassOf")`)

This
- is only minorly more verbose
- gets rid of all the `Def.setting` complexity
- allows you to use the helper all the time (so even in the case here).
- allows you to use it in `mapSourceURISetting`

Reply:
Done.